### PR TITLE
Fix meta serialization

### LINKED ISSUE #158: Meta field in TwirpError should be initialised to empty Map
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/twirphp/twirp/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://github.com/twirphp/twirp/blob/master/.github/SUPPORT.md) without success.

### Version

0375522

### PHP version

8.0

### Go version

_No response_

### Expected Behavior

`meta` field of TwirpError should be initialised as empty Map or other PHP equivalent object.

### Actual Behavior

`meta` field of TwirpError gets initialised as empty list which causes deserialization error in clients. In my case it's java client.
It is happening here: https://github.com/twirphp/twirp/blob/0375522bc04167ec00622cbf65a96caded9ab23e/protoc-gen-twirp_php/templates/global/TwirpError.php.tmpl#L24

### Steps To Reproduce

_No response_

### Additional Information

Given the [spec](https://twitchtv.github.io/twirp/docs/errors.html), meta key should be an optional key => value string pairs. It doesn't mention an empty [] array
```
meta: (optional) key-value pairs with arbitrary string metadata. Useful to define subtypes under the same code, or add extra fields for the callers.
````

### PR DESCRIPTION
**Pull request by sagikazarmark**
**Overview**

Fix spec violation reported in #158

**What this PR does / why we need it**

Fixes #158

### CODE REVIEW

File: protoc-gen-twirp_php/templates/service/Server.php.tmpl [hunk 1]
```diff
@@ -335,7 +335,7 @@ final class {{ .Service | phpServiceName .File }}Server implements RequestHandle
             'code' => $e->getErrorCode(),
             'msg' => $e->getMessage(),
             'meta' => $e->getMetaMap(),
-        ]));
+        ], \JSON_FORCE_OBJECT));
```
Comment (ManInTheBox) [inline protoc-gen-twirp_php/templates/service/Server.php.tmpl:338]: Hey @sagikazarmark although this looks much better now, it still doesn't adhere to the spec https://twitchtv.github.io/twirp/docs/errors.html

> meta: (optional) key-value pairs with arbitrary string metadata. Useful to define subtypes under the same code, or add extra fields for the callers.

The `meta` key should be _optional_. So it should be omitted if it equals to the empty object.

Comment (sagikazarmark) [reply protoc-gen-twirp_php/templates/service/Server.php.tmpl]: Last commit should fix that too

Comment (ManInTheBox) [reply protoc-gen-twirp_php/templates/service/Server.php.tmpl]: I just quickly looked what Twitch's Go impl does, and indeed they're making it optional https://github.com/twitchtv/twirp/blob/4e1d44e644d7399417a9c9273dca725d98e731dd/errors.go#L102

Removing `meta` entirely could probably introduce BC breaks. Although, it's a bug fix to conform to the spec.
Perhaps the safest way is to deprecate the old behavior, bump the version, in the next major version fully remove the key if it's empty.

Not sure what's your release process here guys and if you're ok going straight to BC break.

Comment (sagikazarmark) [reply protoc-gen-twirp_php/templates/service/Server.php.tmpl]: > Removing `meta` entirely could probably introduce BC breaks.

I don't think that's a concern here. The important thing is that clients can interpret what's traveling over the wire.

From what I can tell though, an empty map is going to be virtually the same as not sending it.

Optional in the Go implementation means that it's either not sent at all, or it's nil. Once again, that's virtually the same thing and even if clients are unable to handle the difference, spec conformity tilts the scale to the direction of including this fix sooner than later.

**Comment by ManInTheBox** [comment]
Thanks Márk, I just tested this fix and it worked fine for me! There's only one _potential_ caveat as you've [said](https://github.com/twirphp/twirp/pull/159#discussion_r1603718273)

> The important thing is that clients can interpret what's traveling over the wire.

which could be a problem with _regular_ proto messages, but in the case of an error, the clients usually receive and interpret the errors right away from the server (request/response) so there's no way this removal of `meta` key could break the message contract (like it would be the case with regular proto messages). After all TwirpError is not a _message_ 🤷 

Overall, I'd say let's go merging this :)
Thanks a bunch!

**Pull request merged**